### PR TITLE
helm: cleanup formula

### PR DESCRIPTION
**Pull request by bashofmann**
This modernizes the helm formula to current best practices and fixes the installation of the man page

This is a follow up PR from the discussion in https://github.com/Homebrew/homebrew-core/pull/56373#issuecomment-644609390.

@SMillerDev @dawidd6 

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/helm.rb [hunk 1]
```diff
@@ -13,38 +13,30 @@ class Helm < Formula
...
-      output = Utils.safe_popen_read("SHELL=bash #{bin}/helm completion bash")
-      (bash_completion/"helm").write output
+    mkdir "man1" do
+      system bin/"helm", "docs", "--type", "man"
+      man1.install Dir["*"]
+    end
 
-      output = Utils.safe_popen_read("SHELL=zsh #{bin}/helm completion zsh")
-      (zsh_completion/"_helm").write output
+    output = Utils.popen_read("SHELL=bash #{bin}/helm completion bash")
```
Comment (SMillerDev) [inline Formula/helm.rb:27]: This should be safe_popen_read

Comment (bashofmann) [reply Formula/helm.rb]: Fixed

**Comment by dawidd6** [comment]
This is good. Just bump the revision so people get man pages quicker and squash your commits please.

**Comment by bashofmann** [comment]
`revision 1` is correct (default seems to be 0)?

**Comment by dawidd6** [comment]
Yes it's correct.

**Review by SMillerDev — APPROVED** [review]
Thanks @bashofmann ! Without contributions like yours it'd be impossible to keep homebrew going with the high standards that users have come to expect from the project. You can feel good knowing that you've made the world a tiny bit better for homebrew users around the world! 👍 🎉

**Pull request closed without merging**